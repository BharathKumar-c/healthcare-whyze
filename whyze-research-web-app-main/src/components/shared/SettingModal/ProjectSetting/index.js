import React from 'react';
import PropTypes from 'prop-types';
import { Input, Segaments, Select, Text, TextArea } from '../../../basic';
import { projectSettingMeasurementsOption } from '../../../../constant';
import { projectSettingConstants } from '../../../../constant/ConstantTexts';
import {
  setClinicalTrial,
  setSponsor,
  setDescription,
  setStandards,
  setMeasurement,
} from '../../../../redux/reducer/projectsReducer';
import { useDispatch } from 'react-redux';

export default function ProjectSetting(props) {
  const { data, errors } = props;
  const dispatch = useDispatch();

  return (
    <div>
      <Input
        className="mb"
        lable={projectSettingConstants.clinicalTrial}
        errorText={projectSettingConstants.clinicalTrialError}
        value={data.clinicalTrial}
        handleInputChange={(e) => dispatch(setClinicalTrial(e.target.value))}
        isError={errors.clinicalTrial}
      />
      <Input
        className="mb"
        lable={projectSettingConstants.sponsor}
        errorText={projectSettingConstants.sponsorError}
        value={data.sponsor}
        handleInputChange={(e) => dispatch(setSponsor(e.target.value))}
        isError={errors.sponsor}
      />
      <TextArea
        className="mb"
        lable={projectSettingConstants.description}
        errorText={projectSettingConstants.descriptionError}
        value={data.description}
        maxLength={1000}
        handleInputChange={(e) => dispatch(setDescription(e.target.value))}
        isError={errors.description}
      />
      <div style={{ display: 'flex' }}>
        <Text className="label" text={projectSettingConstants.standardsText} />
        <Text
          className="sub-label"
          text={projectSettingConstants.standardDescription}
        />
      </div>
      <div id="project-setting-select">
        <Select
          className="mb"
          placeholder={projectSettingConstants.rangeLabel}
          options={[]}
          name="range"
          value={data.standards}
          handleSelect={(e) => dispatch(setStandards(e))}
          isError={errors.standards}
          errorText={projectSettingConstants.standardsError}
        />
      </div>
      <div style={{ display: 'flex', flexDirection: 'column' }}>
        <Text
          className="label"
          text={projectSettingConstants.measurementText}
        />
      </div>
      <Segaments
        options={projectSettingMeasurementsOption}
        size={'large'}
        value={data.measurement}
        onChange={(e) => dispatch(setMeasurement(e))}
      />
    </div>
  );
}

ProjectSetting.propTypes = {
  data: PropTypes.object,
  errors: PropTypes.object,
};

ProjectSetting.defalutProps = {
  data: {},
  errors: {},
};
