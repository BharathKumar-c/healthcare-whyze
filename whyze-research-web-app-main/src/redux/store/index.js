import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
