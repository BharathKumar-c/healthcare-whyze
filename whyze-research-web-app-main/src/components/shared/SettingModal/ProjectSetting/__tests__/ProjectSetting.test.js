import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import React from 'react';
import ProjectSetting from '..';
function renderWithContext(element) {
  const { container, getByText, getByRole } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText, getByRole };
}

describe('create test cases for ProjectSettings', () => {
  test('create snapshots for ProjectSetting', () => {
    const { container } = renderWithContext(
      <ProjectSetting handleClickShare={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });
  test('should render text corrctly in tag ', () => {
    const handleclick = jest.fn();
    renderWithContext(<ProjectSetting handleClickShare={handleclick} />);
    const shareSelect = screen.getAllByRole('combobox');
    expect(shareSelect[0]).toBeInTheDocument();
    expect(shareSelect[0]).toHaveAttribute('aria-label', 'Select-option');
    fireEvent.change(shareSelect[0], { target: { value: 'From' } });
    expect(shareSelect[0].value).toBe('From');
    const submitButton = screen.getByTestId('share-button');
    fireEvent.click(submitButton);
    expect(handleclick).toBeCalledTimes(1);
  });
  test('should trigger save button to save the project', () => {
    const handleclick = jest.fn();
    const { container } = renderWithContext(
      <ProjectSetting handleClickShare={handleclick} />,
    );
    const submitButton = container.getElementsByClassName('share-button');
    fireEvent.click(submitButton[0]);
    expect(handleclick).toBeCalledTimes(1);
  });
  test('should check tagRender Corrctly', () => {
    const handleclick = jest.fn();
    const { container } = renderWithContext(
      <ProjectSetting handleClickShare={handleclick} />,
    );
    const shareSelect = screen.getAllByRole('combobox');
    expect(shareSelect[0]).toBeInTheDocument();
    expect(shareSelect[0]).toHaveAttribute('aria-label', 'Select-option');
    fireEvent.click(shareSelect[0]);
    fireEvent.change(shareSelect[0], { target: { value: 'CR' } });

    const option = screen.getByTitle('CR');
    fireEvent.click(option);
    expect(shareSelect[0].value).toBe('');
    const tegRender = container.getElementsByClassName('tag-container');
    expect(tegRender[0]).toBeInTheDocument();
  });
});
