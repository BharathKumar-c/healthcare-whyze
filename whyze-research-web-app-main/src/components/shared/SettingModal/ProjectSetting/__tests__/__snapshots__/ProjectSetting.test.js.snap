// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create test cases for ProjectSettings create snapshots for ProjectSetting 1`] = `
<div>
  <div>
    <div
      class="mb"
    >
      <span
        class="ant-typography input-lable"
      >
        Project
      </span>
      <input
        class="ant-input"
        placeholder=""
        style="border-radius: 0px;"
        type="text"
        value=""
      />
    </div>
    <div
      class="mb"
    >
      <span
        class="ant-typography input-lable"
      >
        Client name
      </span>
      <input
        class="ant-input"
        placeholder=""
        style="border-radius: 0px;"
        type="text"
        value=""
      />
    </div>
    <div
      class="mb"
      id=""
    >
      <span
        class="ant-typography text-area-lable"
      >
        Description
      </span>
      <textarea
        class="ant-input text-area-input"
        name=""
        placeholder=""
      />
    </div>
    <div
      class="alignment"
    >
      <span
        class="ant-typography select-lable"
      >
        Sharing
      </span>
      <div
        aria-label="Select-option"
        class="ant-select mb Select-option ant-select-multiple ant-select-show-search"
        name="range"
      >
        <div
          class="ant-select-selector"
        >
          <div
            class="ant-select-selection-overflow"
          >
            <div
              class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
              style="opacity: 1;"
            >
              <div
                class="ant-select-selection-search"
                style="width: 0px;"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-label="Select-option"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  role="combobox"
                  type="search"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="ant-select-selection-search-mirror"
                >
                  Â 
                </span>
              </div>
            </div>
          </div>
          <span
            class="ant-select-selection-placeholder"
          >
            Select
          </span>
        </div>
      </div>
    </div>
    <button
      class="ant-btn ant-btn-default share-button"
      data-testid="share-button"
      name="Share"
      type="button"
    >
      <svg>
        ShareIcon.svg
      </svg>
      <span>
        Share
      </span>
    </button>
    <div
      style="display: flex;"
    >
      <span
        class="ant-typography label"
      >
        Standards
      </span>
      <span
        class="ant-typography sub-label"
      >
        Standardising markers and value outputs
      </span>
    </div>
    <div
      class="alignment"
    >
      <div
        class="ant-select mb ant-select-single ant-select-show-arrow"
        name="range"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              id="rc_select_TEST_OR_SSR"
              readonly=""
              role="combobox"
              style="opacity: 0;"
              type="search"
              unselectable="on"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-placeholder"
          >
            Select
          </span>
        </div>
        <span
          aria-hidden="true"
          class="ant-select-arrow"
          style="user-select: none;"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      <span
        class="ant-typography label"
      >
        Default measurement
      </span>
    </div>
    <div
      class="ant-segmented ant-segmented-lg"
    >
      <div
        class="ant-segmented-group"
      >
        <label
          class="ant-segmented-item ant-segmented-item-selected"
        >
          <input
            checked=""
            class="ant-segmented-item-input"
            type="radio"
          />
          <div
            class="ant-segmented-item-label"
            title="Metric"
          >
            Metric
          </div>
        </label>
        <label
          class="ant-segmented-item"
        >
          <input
            class="ant-segmented-item-input"
            type="radio"
          />
          <div
            class="ant-segmented-item-label"
            title="Imperial"
          >
            Imperial
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;
